subsys/bluetooth/common/Kconfig

```diff
@@ -77,10 +77,21 @@ config BT_BUF_ACL_RX_SIZE
 	  In a Controller only build this will determine the maximum ACL size
 	  that the Controller will send to the Host.
 
+config BT_BUF_ACL_RX_COUNT_EXTRA
+	int "Number of extra incoming ACL data buffers"
+	default 0
+	range 0 65535
+	help
+	  Number of incoming extra ACL data buffers sent from the Controller to
+	  the Host.
+
+	  By default, the number of incoming ACL data buffers is equal to
+	  CONFIG_BT_MAX_CONN + 1.
+
 config BT_BUF_ACL_RX_COUNT
-	int "Number of incoming ACL data buffers"
-	default 6
-	range 2 256
+	int "[DEPRECATED] Number of incoming ACL data buffers"
```

Comment:
In the commit message you write 

> Because the number of ACL RX buffers must be at least the number of
maximum connections plus one, increasing `CONFIG_BT_MAX_CONN` could
inadvertently lead to a build failure if the number of ACL RX buffers is
not also increased. This dependency may not be obvious to users.

What is the reason why we don't just have `BT_BUF_ACL_RX_COUNT` depend on such a value? The issue seems solvable by Kconfig without deprecating this, and even if it isn't solvable by Kconfig directly (`depends on` can't fix everything), `BUILD_ASSERT`s might be useful to achieve the same result as here.

Reply:
> What is the reason why we don't just have `BT_BUF_ACL_RX_COUNT` depend on such a value?

The problem is that you cannot do something like `depends on CONFIG_XXX + 1`.

> and even if it isn't solvable by Kconfig directly (`depends on` can't fix everything), `BUILD_ASSERT`s might be useful to achieve the same result as here.

We had a build assert but yet, we had some users getting confused. If at some point you set `CONFIG_BT_MAX_CONN` to 6, then you face a build assert because a config that you did not set yourself doesn't have the right value. And then users will have to keep in their config a symbol because the host doesn't handle its dependencies itself.
It feel wrong to me.

Reply:
Yeah, I get that. It's a fair point. 

@jhedberg since we are refactoring this, do you think there's a usecase to use something similar to `HEAP_MEM_POOL_ADD_SIZE` for the ACL buffer counts? Or is that overkill for these buffers?

Reply:
> @jhedberg since we are refactoring this, do you think there's a usecase to use something similar to `HEAP_MEM_POOL_ADD_SIZE` for the ACL buffer counts? Or is that overkill for these buffers?

What do you mean? Are you asking for a way to add more buffers? I added `CONFIG_BT_BUF_RX_COUNT_EXTRA` for that. But I am not sure it's what you're asking 😅

Reply:
> @jhedberg since we are refactoring this, do you think there's a usecase to use something similar to `HEAP_MEM_POOL_ADD_SIZE` for the ACL buffer counts? Or is that overkill for these buffers?

We could do that, but my gut feeling is that it might be overkill, especially if `RX_COUNT_EXTRA` does the job sufficiently well.

Reply:
Oh I see how it work. It require some logic in `kernel/CMakeLists.txt` it's maybe a bit overkill indeed.

Reply:
There's actually some arithmetic operations supported in Kconfig directly. I wonder if that could have been a way to provide a nicer solution: https://docs.zephyrproject.org/latest/build/kconfig/preprocessor-functions.html#integer-functions

Reply:
Oh I wasn't aware. But anyway I think what we choose to do there is better, to me it doesn't make sense to have the user configure such option directly. They can add more if they want but the bare minimum is computed by the Host itself.